### ISSUE TITLE
Unexpected shutdown

### ISSUE THREAD

**Issue by Harlber** [appium/appium#14305]
Unexpected shutdown occurs while running in Pycharm

## Environment

* Appium version 1.15.1
* Appium_Python_Client 0.52
* Desktop OS:Mac
* Appium desktop version:v1.15.1 
* Node.js version v10.10.0
* Mobile platform android 8.0
* Real device HUAWEI STF-AL00

## Details

IDE: Pycharm 2020.1(Community Edition)
Python version: python 3.7

##  Running logs
 ```java
[debug] [ADB] Install command stdout: Success
[debug] [UiAutomator2] Waiting up to 30000ms for services to be available
[debug] [ADB] Running '/Users/hana/Library/Android/sdk/platform-tools/adb -P 5037 -s 8BNDU17909005181 shell pm list instrumentation'
[debug] [UiAutomator2] Instrumentation target 'io.appium.uiautomator2.server.test/androidx.test.runner.AndroidJUnitRunner' is available
[debug] [UiAutomator2] No app capability. Assuming it is already on the device
[debug] [UiAutomator2] Performing shallow cleanup of automation leftovers
[debug] [UiAutomator2] No obsolete sessions have been detected (Error: read ECONNRESET)
[debug] [ADB] Running '/Users/hana/Library/Android/sdk/platform-tools/adb -P 5037 -s 8BNDU17909005181 shell am force-stop io.appium.uiautomator2.server.test'
[UiAutomator2] Starting UIAutomator2 server 4.8.1
[UiAutomator2] Using UIAutomator2 server from '/usr/local/lib/node_modules/appium/node_modules/_appium-uiautomator2-server@4.8.1@appium-uiautomator2-server/apks/appium-uiautomator2-server-v4.8.1.apk' and test from '/usr/local/lib/node_modules/appium/node_modules/_appium-uiautomator2-server@4.8.1@appium-uiautomator2-server/apks/appium-uiautomator2-server-debug-androidTest.apk'
[UiAutomator2] Waiting up to 30000ms for UiAutomator2 to be online...
[debug] [ADB] Creating ADB subprocess with args: ["-P",5037,"-s","8BNDU17909005181","shell","am","instrument","-w","io.appium.uiautomator2.server.test/androidx.test.runner.AndroidJUnitRunner"]
[debug] [Instrumentation] io.appium.uiautomator2.server.test.AppiumUiAutomator2Server:
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8200/wd/hub/status] with no body
[WD Proxy] Got response with unknown status: {"errno":"ECONNRESET","code":"ECONNRESET","syscall":"read"}
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8200/wd/hub/status] with no body
[WD Proxy] Got response with unknown status: {"errno":"ECONNRESET","code":"ECONNRESET","syscall":"read"}
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8200/wd/hub/status] with no body
[debug] [WD Proxy] Got response with status 200: {"sessionId":"None","value":{"message":"UiAutomator2 Server is ready to accept commands","ready":true}}
[debug] [UiAutomator2] The initialization of the instrumentation process took 3214ms
[debug] [WD Proxy] Matched '/session' to command name 'createSession'
[debug] [WD Proxy] Proxying [POST /session] to [POST http://127.0.0.1:8200/wd/hub/session] with body: {"capabilities":{"firstMatch":[{"platform":"LINUX","webStorageEnabled":false,"takesScreenshot":true,"javascriptEnabled":true,"databaseEnabled":false,"networkConnectionEnabled":true,"locationContextEnabled":false,"warnings":{},"desired":{"platformName":"Android","platformVersion":"8.0","deviceName":"HUAWEI STF-AL00","appPackage":"cn.caocaokeji.user","appActivity":"cn.caocaokeji.compat.load.LoadActivity","noReset":true,"udid":"8BNDU17909005181","autoGrantPermissons":true,"autoLaunch":true,"newCommandTimeout":600000},"platformName":"Android","platformVersion":"8.0","deviceName":"8BNDU17909005181","appPackage":"cn.caocaokeji.user","appActivity":"cn.caocaokeji.compat.load.LoadActivity","noReset":true,"udid":"8BNDU17909005181","autoGrantPermissons":true,"autoLaunch":true,"newCommandTimeout":600000,"deviceUDID":"8BNDU17909005181"}],"alwaysMatch":{}}}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"440059f0-ccd7-4e0d-8bb2-d36fee47726f","value":{"capabilities":{"firstMatch":[{"platform":"LINUX","webStorageEnabled":false,"takesScreenshot":true,"javascriptEnabled":true,"databaseEnabled":false,"networkConnectionEnabled":true,"locationContextEnabled":false,"warnings":{},"desired":{"platformName":"Android","platformVersion":"8.0","deviceName":"HUAWEI STF-AL00","appPackage":"cn.caocaokeji.user","appActivity":"cn.caocaokeji.compat.load.LoadActivity","noReset":true,"udid":"8BNDU17909005181","autoGrantPermissons":true,"autoLaunch":true,"newCommandTimeout":600000},"platformName":"Android","platformVersion":"8.0","deviceName":"8BNDU17909005181","appPackage":"cn.caocaokeji.user","appActivity":"cn.caocaokeji.compat.load.LoadActivity","noReset":true,"udid":"8BNDU17909005181","autoGrantPermissons":true,"autoLaunch":true,"newCommandTimeout":600000,"deviceUDID":"8BNDU17909005181"}],"alwaysMatch":{}},"sessionId":"440059f0-ccd7-4e0d-8bb2-d36fee47726f"}}
[WD Proxy] Determined the downstream protocol as 'W3C'
[debug] [WD Proxy] Proxying [GET /appium/device/info] to [GET http://127.0.0.1:8200/wd/hub/session/440059f0-ccd7-4e0d-8bb2-d36fee47726f/appium/device/info] with no body
[debug] [WD Proxy] Got response with status 200: {"sessionId":"440059f0-ccd7-4e0d-8bb2-d36fee47726f","value":{"androidId":"be2ea2fcb3007ef6","apiVersion":"26","bluetooth":{"state":"OFF"},"brand":"HONOR","carrierName":"中国电信","displayDensity":480,"locale":"zh_CN_#Hans","manufacturer":"HUAWEI","model":"STF-AL00","networks":[{"capabilities":{"SSID":null,"linkDownBandwidthKbps":1048576,"linkUpstreamBandwidthKbps":1048576,"networkCapabilities":"NET_CAPABILITY_NOT_METERED,NET_CAPABILITY_INTERNET,NET_CAPABILITY_NOT_RESTRICTED,NET_CAPABILITY_TRUSTED,NET_CAPABILITY_NOT_VPN,NET_CAPABILITY_VALIDATED,NET_CAPABILITY_NOT_ROAMING","signalStrength":-64,"transportTypes":"TRANSPORT_WIFI"},"detailedState":"CONNECTED","extraInfo":"\"Geely-出行研究院-5G\"","isAvailable":true,"isConnected":true,"isFailover":false,"isRoaming":false,"state":"CONNECTED","subtype":0,"subtypeName":"","type":1,"typeName":"WIFI"}],"platformVersion":"8.0.0","realDisplaySize":"1080x1920","timeZone":"Asia/Shanghai"}}
[debug] [ADB] Running '/Users/hana/Library/Android/sdk/platform-tools/adb -P 5037 -s 8BNDU17909005181 shell dumpsys window'
[AndroidDriver] Screen already unlocked, doing nothing
[UiAutomator2] Starting 'cn.caocaokeji.user/cn.caocaokeji.compat.load.LoadActivity and waiting for 'cn.caocaokeji.user/cn.caocaokeji.compat.load.LoadActivity'
[debug] [ADB] Running '/Users/hana/Library/Android/sdk/platform-tools/adb -P 5037 -s 8BNDU17909005181 shell am start -W -n cn.caocaokeji.user/cn.caocaokeji.compat.load.LoadActivity -S -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -f 0x10200000'
[debug] [WD Proxy] Proxying [GET /appium/device/pixel_ratio] to [GET http://127.0.0.1:8200/wd/hub/session/440059f0-ccd7-4e0d-8bb2-d36fee47726f/appium/device/pixel_ratio] with body: {}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"440059f0-ccd7-4e0d-8bb2-d36fee47726f","value":3.0}
[debug] [WD Proxy] Matched '/appium/device/system_bars' to command name 'getSystemBars'
[debug] [WD Proxy] Proxying [GET /appium/device/system_bars] to [GET http://127.0.0.1:8200/wd/hub/session/440059f0-ccd7-4e0d-8bb2-d36fee47726f/appium/device/system_bars] with body: {}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"440059f0-ccd7-4e0d-8bb2-d36fee47726f","value":{"statusBar":72}}
[debug] [WD Proxy] Matched '/window/current/size' to command name 'getWindowSize'
[debug] [WD Proxy] Proxying [GET /window/current/size] to [GET http://127.0.0.1:8200/wd/hub/session/440059f0-ccd7-4e0d-8bb2-d36fee47726f/window/current/size] with body: {}
[debug] [WD Proxy] Got response with status 200: {"sessionId":"440059f0-ccd7-4e0d-8bb2-d36fee47726f","value":{"height":1920,"width":1080}}
[Appium] New AndroidUiautomator2Driver session created successfully, session 86012b42-9db1-46cc-b532-21f4fcc21196 added to master session list
[Appium] Closing session, cause was 'Unexpected shutdown'
[Appium] Removing session 86012b42-9db1-46cc-b532-21f4fcc21196 from our master session list
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1588909195078 (11:39:55 GMT+0800 (China Standard Time))
[debug] [W3C (86012b42)] Cached the protocol value 'W3C' for the new session 86012b42-9db1-46cc-b532-21f4fcc21196
.
.
.
[HTTP] --> POST /wd/hub/session/86012b42-9db1-46cc-b532-21f4fcc21196/element
[HTTP] {"using":"id","value":"cn.caocaokeji.user:id/platform_home_top_city_click"}
[debug] [W3C (86012b42)] Encountered internal error running command: NoSuchDriverError: A session is either terminated or not started
[debug] [W3C (86012b42)]     at asyncHandler (/usr/local/lib/node_modules/appium/node_modules/_appium-base-driver@4.5.1@appium-base-driver/lib/protocol/protocol.js:252:15)
[debug] [W3C (86012b42)]     at app.(anonymous function) (/usr/local/lib/node_modules/appium/node_modules/_appium-base-driver@4.5.1@appium-base-driver/lib/protocol/protocol.js:432:15)
[debug] [W3C (86012b42)]     at Layer.handle [as handle_request] (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/layer.js:95:5)
[debug] [W3C (86012b42)]     at next (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/route.js:137:13)
[debug] [W3C (86012b42)]     at Route.dispatch (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/route.js:112:3)
[debug] [W3C (86012b42)]     at Layer.handle [as handle_request] (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/layer.js:95:5)
[debug] [W3C (86012b42)]     at /usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:281:22
[debug] [W3C (86012b42)]     at param (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:354:14)
[debug] [W3C (86012b42)]     at param (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:365:14)
[debug] [W3C (86012b42)]     at Function.process_params (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:410:3)
[debug] [W3C (86012b42)]     at next (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:275:10)
[debug] [W3C (86012b42)]     at logger (/usr/local/lib/node_modules/appium/node_modules/_morgan@1.10.0@morgan/index.js:144:5)
[debug] [W3C (86012b42)]     at Layer.handle [as handle_request] (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/layer.js:95:5)
[debug] [W3C (86012b42)]     at trim_prefix (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:317:13)
[debug] [W3C (86012b42)]     at /usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:284:7
[debug] [W3C (86012b42)]     at Function.process_params (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:335:12)
[debug] [W3C (86012b42)]     at next (/usr/local/lib/node_modules/appium/node_modules/_express@4.17.1@express/lib/router/index.js:275:10)
[debug] [W3C (86012b42)]     at /usr/local/lib/node_modules/appium/node_modules/_body-parser@1.19.0@body-parser/lib/read.js:130:5
[debug] [W3C (86012b42)]     at invokeCallback (/usr/local/lib/node_modules/appium/node_modules/_raw-body@2.4.0@raw-body/index.js:224:16)
[debug] [W3C (86012b42)]     at done (/usr/local/lib/node_modules/appium/node_modules/_raw-body@2.4.0@raw-body/index.js:213:7)
[debug] [W3C (86012b42)]     at IncomingMessage.onEnd (/usr/local/lib/node_modules/appium/node_modules/_raw-body@2.4.0@raw-body/index.js:273:7)
[debug] [W3C (86012b42)]     at IncomingMessage.emit (events.js:182:13)
[HTTP] <-- POST /wd/hub/session/86012b42-9db1-46cc-b532-21f4fcc21196/element 404 24
```

**Comment by mykola-mokhnach** [comment]
Please provide the full log and update the server to the most recent version

**Comment by Harlber** [comment]
Solved by upgrade appium(v1.15.1 -> v1.17.1)!
Thanks a lot @mykola-mokhnach
